doc/getting-started.md

```diff
@@ -0,0 +1,583 @@
+
+The complete echo server can be found
+[here](../examples/ping_pong/src/main.rs).
+
+> **Note:** As of the time of writing, Mio does not support Windows
+> (though support is currently in progress).
+
+Before you get started, setup an empty text file. Write down any
+thoughts, confusions, questions that you have while going through this
+guide. You only have one first impression, and I would like to capture
+this in order to improve this document.
+
+Post your notes as a comment [here](https://github.com/carllerche/mio/pull/222).
+
+## Setting up the project
+
+The first step is getting a new Cargo project setup. In a new
+directory, run the following:
+
+```not_rust
+cargo new pingpong --bin
+cd pingpong
+```
+
+Now, open the directory in your favorite text editor. You should see the
+following files:
+
+* src/main.rs
+* Cargo.toml
+
+If you are not already familiar with Cargo, you can learn more about it
+[here](http://doc.crates.io/).
+
+Open `Cargo.toml` and add a dependency on Mio by putting the following a
+the bottom of the file:
+
+```toml
+[dependencies]
+mio = "0.4.1"
+```
+
+Save the file, then compile and run the project using the following
+command:
+
+```not_rust
+cargo run
+```
+
+You will see some Cargo related output followed by `Hello, world!`. We
+haven't written any code yet and this is the default behavior of a
+freshly generated Cargo project.
+
+## Writing the Echo Server
+
+Let's start by writing a very simple server that accepts connections and
+does nothing with them. The connections will be accepted and shutdown
+immediately after.
+
+Here is the entire code, we'll step through it in a bit.
+
+```rust
+extern crate mio;
+
+use mio::tcp::*;
+
+const SERVER: mio::Token = mio::Token(0);
+
+struct Pong {
+    server: TcpListener,
+}
+
+impl mio::Handler for Pong {
+    type Timeout = ();
+    type Message = ();
+
+    fn ready(&mut self, event_loop: &mut mio::EventLoop<Pong>, token: mio::Token, events: mio::EventSet) {
+        match token {
+            SERVER => {
+                // Only receive readable events
+                assert!(events.is_readable());
+
+                println!("the server socket is ready to accept a connection");
+                match self.server.accept() {
+                    Ok(Some(socket)) => {
+                        println!("accepted a socket, exiting program");
+                        event_loop.shutdown();
+                    }
+                    Ok(None) => {
+                        println!("the server socket wasn't actually ready");
+                    }
+                    Err(e) => {
+                        println!("listener.accept() errored: {}", e);
+                        event_loop.shutdown();
+                    }
+                }
+            }
+            _ => panic!("Received unknown token"),
+        }
+    }
+}
+
+fn main() {
+    let address = "0.0.0.0:6567".parse().unwrap();
+    let server = TcpListener::bind(&address).unwrap();
```

Comment:
I found in other project this syntax of error handling to be quite elegant rather than just an `unwrap` 

```rust
 let sock = TcpListener::bind(&addr).ok().expect("Failed to bind address");
```